text/0097-log-data-model.md

```diff
@@ -0,0 +1,971 @@
+field equal to INFO then it may be preferable from the user experience
+perspective to ensure that **severity="Informational"** and **severity="INFO"**
+conditions both to are TRUE for that record.
+
+### Field: `ShortName`
+
+Type: string.
+
+Description: Short event identifier that does not contain varying parts.
+`ShortName` describes what happened (e.g. "ProcessStarted"). Recommended to be
+no longer than 50 characters. Optional. Not guaranteed to be unique in any way.
+Typically used for filtering and grouping purposes in backends.
+
+### Field: `Body`
+
+Type: any.
+
+Description: A value containing the body of the log record (see the description
+of `any` type above). Can be for example a human-readable string message
+(including multi-line) describing the event in a free form or it can be a
+structured data composed of arrays and maps of other values. Can vary for each
+occurrence of the event coming from the same source.
+
+### Field: `Resource`
+
+Type: key/value pair list.
+
+Description: Describes the source of the log, aka
+[resource](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#resources).
+"value" of each pair is of `any` type. Multiple occurrences of events coming
+from the same event source can happen across time and they all have the same
+value of `Resource`. Can contain for example information about the application
+that emits the record or about the infrastructure where the application runs.
+Data formats that represent this data model may be designed in a manner that
+allows the `Resource` field to be recorded only once per batch of log records
+that come from the same source. SHOULD follow OpenTelemetry
+[semantic conventions for Resources](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions).
+
+### Field: `Attributes`
+
+Type: key/value pair list.
+
+Description: Additional information about the specific event occurrence. "value"
+of each pair is of `any` type. Unlike the `Resource` field, which is fixed for a
+particular source, `Attributes` can vary for each occurrence of the event coming
+from the same source. Can contain information about the request context (other
+than TraceId/SpanId). SHOULD follow OpenTelemetry
+[semantic conventions for Attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions).
+
+## Example Log Records
+
+Below are examples that show one possible representation of log records in JSON.
+These are just examples to help understand the data model. Don’t treat the
+examples as _the_ way to represent this data model in JSON.
+
+This document does not define the actual encoding and format of the log record
+representation. Format definitions will be done in separate OTEPs (e.g the log
+records may be represented as msgpack, JSON, Protocol Buffer messages, etc).
+
+Example 1
```

Comment:
Here is an extension of Example 1 with more details about the Kubernetes deployment.

Differences with the proposed structure:
* I used namespaces as sub grouping elements (e.g. `resource/helm_chart`, `resource/container`, `resource/k8s`,  `resource/cloud`...)
* I pushed down inside `attributes/log` the `Severity`
* I pushed `trace_id` and `span_id` down in a `trace_context` top-level field as suggested by @reyang in https://github.com/open-telemetry/oteps/pull/97/files#r418803900

```

├── timestamp: 1586960586000
├── body: 20200415T072306-0700 INFO Donut order placed
├── resource
│   ├── service
│   │   ├── name: donut-shop
│   │   ├── feature_flags: ["anti_fraud_2020_04_01", "cross_selling_2020_04_12"]
│   │   └── version: 2.0.0
│   ├── helm_chart
│   │   └── name: donut-shop
│   │   └── app_version: 2.0.0
│   │   └── version: 2.0.0
│   ├── container
│   │   ├── image.name: docker.mycie.com/donut_shop
│   │   └── image.tag: 2.0.0
│   ├── k8s
│   │   └── cluster.name: website-production
│   │   └── deployment.name: production-donut-shop
│   │   └── pod.name: production-donut-shop-7758b9cdf8-cs798
│   │   └── service.name: production-donut-shop
│   │   └── service.external_ip:  # <none>
│   ├── host
│   │   └── id: i-1234567890abcdef0
│   │   └── ip: 10.101.163.102
│   ├── cloud
│   │   └── zone: us-east-1c
│   │   └── region: us-east-1
│   ├── labels
│   │   └── my.custom.application.tag: hello
├── trace_context
│   ├── trace_id: f4dbb3edd765f620
│   ├── span_id: 43222c2d51a7abe3
├── attributes
│   ├── log
│   │   ├── level: INFO
│   │   ├── logger: com.mycie.donut.shop
│   │   └── thread: http-123
    ├── http
    │   ├── scheme: https
    │   ├── host: donut.mycie.com
    │   ├── target:/order
    │   ├── method: post
    │   └── client_ip: 192.0.2.5
```

Reply:
Resolving. Feel free to reopen if needed.